---
parent: INI
ancestor: Rules
permalink: /rules/Proj4002
---

# Proj4002: Invalid INI key-value pair
An [INI file](https://en.wikipedia.org/wiki/INI_file) is a plain text file with
a structure and syntax comprising keyâ€“value pairs organized in sections.

Key-value pairs can either be separated by `=` or `:`.

## Non-compliant
``` ini
Key1 =

Key2 == value

= value

Key4 value
```

## Compliant
``` ini
Key1 = value

Key2 = value

Key3 = value

Key4 = value
```