---
parent: INI
ancestor: Rules
---

# Proj4001: Invalid INI Header
An [INI file](https://en.wikipedia.org/wiki/INI_file) is a plain text file with
a structure and syntax comprising keyâ€“value pairs organized in sections.

Section headers start with a `[` and end with a `]`.

## Non-compliant
``` ini
[Header1

[Header2]]

[[Header3]

[]
```

## Compliant
``` ini
[Header1]

[Header2]

[Header3]

[Header4]
```