---
parent: Packaging
ancestor: Rules
---

# Proj0218: Symbol package format snupkg requires debug type portable
[When creating symbols `.snupkg` package](https://learn.microsoft.com/nuget/create-packages/symbol-packages-snupkg),
having `<DebugType>` set to `full` or `pdbonly` results in an invalid `.snupkg`
(or at least one that the official nuget symbol server can't process). Meanwhile,
`<DebugType>` value `embedded` results in the symbols being included in the
 regular binaries, defeating the purpose of the separate `.snupkg` file.

## Non-compliant
``` xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <SymbolPackageFormat>snupkg</SymbolPackageFormat>
    <DebugType>embedded</DebugType>
  </PropertyGroup>

</Project>
```

## Compliant
``` xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <SymbolPackageFormat>snupkg</SymbolPackageFormat>
    <DebugType>portable</DebugType>
  </PropertyGroup>

</Project>
```
