---
parent: General
ancestor: MSBuild
---

# Proj0038: Fully specify NoWarn rule IDs
MSBuild allows disabling a static code analysis rule by adding it to the
`<NoWarn>` property. It allows rule IDs that are only specified by their
numeric part. This is considered a bad practice: it can lead to excluding
multiple rules at once that have the same integer part, and makes it less
trivial to understand which rule has been disabled.

## Non-compliant
``` xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <NoWarn>1825</NoWarn>
  </PropertyGroup>

</Project>
```

## Compliant
``` xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <NoWarn>CA1825</NoWarn>
  </PropertyGroup>

</Project>
```
